Best practices/React/Organization.md

```diff
@@ -4,11 +4,111 @@
 
 This section describes a sensible starting point for organizing a React application that supports server-side rendering.
 
-## `/app`
+## Application
 
-### `/app/components` and `<Component>/components`
+### React Components and Sub-components
+
+React Components should always exist in a `components/` folder, but the component's internal structure may change depending on their complexity.
+
+
+#### 1. Start with a file and a test
+
+At its simplest a component will exist as a component and a test at the `components/` level.
```

Comment:
Should we recommend starting with an `index.ts` that exports the component right off the bat?